Kernel/RobSubstitution.hpp

```diff
@@ -17,55 +17,331 @@
 #define __RobSubstitution__
 
 #include "Forwards.hpp"
+#include "Kernel/Polynomial.hpp"
 #include "Lib/Backtrackable.hpp"
+#include "Lib/Recycled.hpp"
 #include "Term.hpp"
-#include "MismatchHandler.hpp"
 #include "Lib/Hash.hpp"
 #include "Lib/DHMap.hpp"
+#include "Lib/Metaiterators.hpp"
+#include "Kernel/BottomUpEvaluation.hpp"
+#include "Lib/Environment.hpp"
+#include "Kernel/Signature.hpp"
+#include "Kernel/TypedTermList.hpp"
+#include <initializer_list>
 
 #if VDEBUG
 #include <iostream>
 #include "Lib/VString.hpp"
 #endif
 
+
+const int GLUE_INDEX=-3;
+const int SPECIAL_INDEX=-2;
+const int UNBOUND_INDEX=-1;
 namespace Kernel
 {
+struct VarSpec
+{
+  VarSpec() {}
```

Comment:
Alas not because `DHMap::Entry` calls default constructors. I thought about changing that because it's often quite annoying but i wasn't arsed (yet) because it's quite some effort.